src/_guides/language/effective-dart/design.md

```diff
@@ -952,13 +975,76 @@ class CallChainVisitor {
 }
 {% endprettify %}
 
+This guideline is deliberately softer than the previous one because the benefits
+of static typing are slightly reduced when code is confined within a single
+library. In practice, though, most Dart users type annotate all declarations,
+public or private.
+
+
+### AVOID annotating types for initialized local variables.
+
+Function bodies in modern code tend to be short, and the types of local
+variables are almost always obvious and inferrable from the initializing
+expression. Writing that type down is often simply visual noise. Instead, spend
+that energy choosing a great *name* for your local variable.
+
+{:.good-style}
+<?code-excerpt "misc/lib/effective_dart/design_good.dart (omit-types-on-locals)"?>
+{% prettify dart %}
+List<List<Ingredient>> possibleDesserts(Set<Ingredient> pantry) {
+  var desserts = <List<Ingredient>>[];
```

Comment:
Make this `final`?

Reply:
We don't have any guidelines around using `final` versus `var` for locals. I think more code I see in reviews uses `var` than `final`, but I'm not sure.

Reply:
@kevmoo might have an opinion on this (Kevin hadn't you suggested, at some point, that we needed a guideline about `final`)?